### DISCUSSION TITLE
Accessing Repos by Model Type or By String

### DISCUSSION THREAD

**Discussion by abbas-alshafai** [flutterdata/flutter_data#137] [Q&A]
Hello!

Let's say I have a model called `User`. The docs say that I can access its repo by the generated `ref.users` or by `ref.watch(usersRepositoryProvider)`. Is there a way to access repos by specifying model types or by passing a string (for the table/collection/box). For example, ObjectBox db does something like:

`
final userBox = store.box<User>()
`

Or maybe something similar to Hive db, where a string is passed to specify the box:
`
final box = await Hive.openBox('users')
`

Thanks in advance!

**Comment by frank06** [comment]
Yes it's possible.

You can use the generated `repositoryProviders` on `main.data.dart`, like

```dart
final repo = ref.watch(repositoryProviders['users']) as Repository<User>;
```

Let me know if that works
